subsys/bluetooth/host/cs.c

```diff
@@ -0,0 +1,49 @@
+/* cs.c - Bluetooth Channel Sounding handling */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/cs.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/hci_types.h>
+
+#include "conn_internal.h"
+
+#if defined(CONFIG_BT_CHANNEL_SOUNDING)
+int bt_cs_set_default_settings(struct bt_conn *conn,
+			       const struct bt_cs_set_default_settings_param *params)
+{
+	struct bt_hci_cp_le_cs_set_default_settings *cp;
+	struct net_buf *buf;
+
+	if (params->max_tx_power > BT_HCI_OP_LE_CS_MAX_MAX_TX_POWER ||
```

Comment:
I think the controller also has to check this, and returns "invalid command parameters" in this case. So we could choose to leave this out

Reply:
yeah good point, no need to duplicate the logic